### PR TITLE
/set and /clear can be used just by admins

### LINKED ISSUE #2: /set and /clear can be done just by admins
Use this to get the admins https://core.telegram.org/bots/api#getchatadministrators

### PR DESCRIPTION
**Pull request by wistcc**
This closes #2

### CODE REVIEW

File: config.js [hunk 1]
```diff
@@ -1,5 +1,13 @@
...
+
+module.exports = {
+  firebase: {
+    apiKey: process.env.FIREBASE_API_KEY,
+    authDomain: 'pingchannelbot.firebaseapp.com',
+    databaseURL: 'https://pingchannelbot.firebaseio.com',
+    storageBucket: 'pingchannelbot.appspot.com',
+  },
+  apiId: process.env.API_ID,
+  currentPort: process.env.PORT,
```
Comment (nramirez) [inline config.js:11]: I would move ``|| 3000`  then we will only have this responsibility in one place.

File: index.js [hunk 2]
```diff
@@ -55,64 +33,138 @@ app.post('/new-message', (req, res) => {
     ref.once('value').then(snapshot => {
       const chat = snapshot.val();
       const usernames = chat && chat.usernames ? chat.usernames : '';
-      console.log('le usernames', chat, usernames);
+
       if (chat && chat.message_id && chat.message_id > message.message_id) {
         console.log('discarting old message', chat.message_id, message.message_id);
-        return res.end('Error: message is undefined');
+        return res.end('discarting old message', chat.message_id, message.message_id);
```
Comment (nramirez) [inline index.js:39]: Minor,  but something like`Discarding... This is an older message`, sounds better.

File: index.js [hunk 3]
```diff
@@ -55,64 +33,138 @@ app.post('/new-message', (req, res) => {
...
   } catch (err) {
-    console.log('Error somewhere: ', err);
-    talkToBot(chatId, `Sorry we didn't get that`, res);
-    res.end('Error somewhere: ', err);
-  }
-
-  res.end('All done!');
+    telegramManager.talkToBot(chatId, `Sorry we didn't get that`).then(() => {
+      console.log('Error somewhere: ', err);
+      res.end('Error somewhere: ', err);
```
Comment (nramirez) [inline index.js:60]: Do we really need to log here? or remove `then` and `err` here and log, anyways you have to log the error on L57

File: index.js [hunk 4]
```diff
@@ -55,64 +33,138 @@ app.post('/new-message', (req, res) => {
...
 
 app.get('/', (req, res) => {
   res.send('Bot up and running!');
 });
 
 // Finally, start our server
-app.listen(process.env.PORT || 3000, () => {
-  console.log(`Telegram app listening on port 3000! Api Key ${process.env.API_ID}`);
+app.listen(config.currentPort || 3000, () => {
+  console.log(`Telegram app listening on port 3000! Api Key ${config.apiId}`);
```
Comment (nramirez) [inline index.js:74]: if you move the Port to the config file, will need to modify this line

File: index.js [hunk 5]
```diff
@@ -55,64 +33,138 @@ app.post('/new-message', (req, res) => {
...
+
+const isAdmin = (admins, user) => admins.filter(admin => admin.user.username === user).length > 0;
+
+const setUsernames = (message, chatId, usernames, ref, res) => {
+  if(message.from) {
+    telegramManager.getChatAdministrators(chatId).then(({ data }) => {
+      if (isAdmin(data.result, message.from.username)) {
+        ref.update({
+          'message_id': message.message_id,
+          'usernames': '' + messageManager.extractUniqueUsernames(message, usernames)
```
Comment (nramirez) [inline index.js:85]: I think you can remove the single quotes for these two

File: index.js [hunk 6]
```diff
@@ -55,64 +33,138 @@ app.post('/new-message', (req, res) => {
...
+    });
+  } else {
+    console.log('Error message.from is undefined');
+    res.end('Error message.from is undefined');
+  }
+};
+
+const pingAll = (chatId, text, res) => {
+  telegramManager.talkToBot(chatId, text).then(() => {
+    res.end('Message posted: ');
```
Comment (nramirez) [inline index.js:117]: minor: missing, `text` here

File: index.js [hunk 7]
```diff
@@ -55,64 +33,138 @@ app.post('/new-message', (req, res) => {
...
+          console.log(err);
+          res.end('Error: ', err);
+        });
+      }
+    }, err => {
+      console.log(err);
+      res.end('Error: ', err);
+    });
+  } else {
+    console.log('Error message.from is undefined:', err);
```
Comment (nramirez) [inline index.js:150]: where does `err` comes from? Looks like this will fail.

File: index.js [hunk 8]
```diff
@@ -55,64 +33,138 @@ app.post('/new-message', (req, res) => {
...
+    res.end('Error message.from is undefined:', err);
+  }
+};
+
+const addOrRemoveParticipant = (message, chatId, usernames, ref, res) => {
+  ref.set({
+    'usernames': messageManager.extractUniqueUsernames(message, usernames),
```
Comment (nramirez) [inline index.js:157]: do you like `trailing commas`? it's annoying to me on JS files

File: index.js [hunk 9]
```diff
@@ -55,64 +33,138 @@ app.post('/new-message', (req, res) => {
...
+  });
+
+  const msg = message.new_chat_participant ?
+    `${message.new_chat_participant.username} was added to the list.` :
+    `${message.left_chat_participant.username} was removed from the list.`;
+
+  telegramManager.talkToBot(chatId, msg).then(() => {
+    res.end(`New or left chat participant managed, username: ${usernames}`);
```
Comment (nramirez) [inline index.js:165]: minor but I think you can do `msg` here.

File: telegramManager.js [hunk 10]
```diff
@@ -0,0 +1,20 @@
+const axios = require('axios');
+const config = require('./config');
+
+const getChatAdministrators = (chatId) => {
```
Comment (nramirez) [inline telegramManager.js:4]: can remove parens

File: telegramManager.js [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+  return axios.post(`${config.telegramBaseUrl}/getChatAdministrators`, {
+    chat_id: chatId,
+  });
+};
+
+const talkToBot = (chatId, text) => {
+  return axios.post(`${config.telegramBaseUrl}/sendMessage`, {
+    chat_id: chatId,
+    text: text
```
Comment (nramirez) [inline telegramManager.js:13]: `trailing` or `no trailing`?

**Review by nramirez — APPROVED** [review]
I left some comments, please address those that you think need to be addressed, and :shipit:

**Comment by nramirez** [comment]
Also, can you please, squash your commits, into meaningful commits? `Typo` doesn't look like something that could help.

File: index.js (hunk 8 above)
Comment (wistcc) [reply index.js:157]: I'm kind of used to it since we are using it on chess.

File: telegramManager.js (hunk 11 above)
Comment (wistcc) [reply telegramManager.js:13]: to-be-or-not-to-be-that-is-the-question

**Pull request merged**